### PR TITLE
Cleaned Up Code Semantics Across Solution

### PR DESCRIPTION
**Pull request by darkmastermindz**
- Cleaned up style
- Cleaned up weird looking code
- Simplified overcomplex Null Checks
- Lowered Code Complexity
- Tightened Scope / Fixed scope Ambiguity issues.
- Fixed Provider Issues
- Fixed Errors Buildable Code

### CODE REVIEW

**Comment by edyoung** [comment]
Hi,

First, I really appreciate that you've taken the time to contribute to OneGet. Thank you!

Second, I have a small problem - this is now a large diff (+16K, -12K), which mixes together some large-scale reformatting with some changes to behavior (at least, that's what "cleaning up semantics" implies to me). I've generally found that large changes need to be reviewed and merged with caution to avoid regressions. Which means it could take a while and multiple rounds of review feedback before we could merge this.

It would be safer from my point of view to consume a set of more targeted PRs, separating 'cleanliness' changes (which only affect formatting); refactoring changes (which rearrange the code but should have no effect on behavior) and semantic changes (this method does the wrong thing in some cases and I have fixed it). 

This article provides a good explanation of the 'perfect pull request':
 https://medium.com/@hugooodias/the-anatomy-of-a-perfect-pull-request-567382bb6067

**Comment by Smorgan05** [comment]
Ok, so what's the thought on this?

-Morgan

**Comment by edyoung** [comment]
You are of course free to make whatever changes you want to your fork. But bear in mind that package management is installed on millions of computers and is used in production by a lot of people in a lot of scenarios. To reduce the risk of regression we can only merge PRs which are smaller and more targeted than this one. If you continue to diverge in a separate branch, the chances we'd be able to merge the whole thing is slim. I believe the better course is to make changes incrementally and send PRs for those.

**Comment by Smorgan05** [comment]
@edyoung,

I am in full agreement with @darkmastermindz in terms of implementing changes to OneGet across the board.  Some of these changes can be facilitated by separating them into their respective PR.  However the semantic changes implemented using machine learning should be put into a single PR.  These changes are non functional and simply serve to implement standardization in terms of code formatting (see what @darkmastermindz stated above).

In regards to package management being installed on millions of production machines the changes being done here is simple refactoring.  Along with targeted changes to code so that areas can be simplified.  These changes are necessary in order to bring Package Managements code quality up to par.

-Morgan

**Comment by edyoung** [comment]
Any pull request needs to have a description which explains what it does, and it should do one thing, or one kind of thing.

I'd consider a PR which reformats whitespace or brace position consistently across every file, even if it touches a large number of lines. 

I'd happily merge a PR which takes advantage of newer language features like => and .?

I'd consider a PR which does a straightforward refactoring (like changing some nested if's to guard clauses) across a number of files. 

I'd consider a PR which fixes a bug or a class of bugs.

I'd welcome a change which fixes an actual security bug with open arms (though those should generally be reported via https://portal.msrc.microsoft.com/en-us/engage/sbr so disclosure can be coordinated).

There are definitely worthwhile changes here. But the reason I am uncomfortable with this change, and would be even less comfortable with a bigger one, is that according to your description, it mixes all of these categories together, and the reviewer is left to puzzle out which change does what. If there are "many security issues" fixed somewhere in here it's unclear where, and any such change needs very careful review. 

I'm also curious what machine learning has to do with any of this?

**Comment by Smorgan05** [comment]
@edyoung,

Ok Makes sense.  We are taking this over to trello to break apart the categories that you have listed in order to make this happen.  I am in the process of going through the entire project to identify overly complex coding using Code Metrics. My game plan is to go through all the code in the project and identify areas that are poorly coded and so forth.  That said I am beginning to get worried about the overall quality of the code across the board.  The code quality is poor.

-Morgan

File: src/ConsoleApp1/ConsoleApp1.csproj [hunk 1]
```diff
@@ -0,0 +1,82 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (bergmeister) [inline src/ConsoleApp1/ConsoleApp1.csproj]: This is an old style csproj that should not be used for new code any more. One should use the `dotnet` command line from the .Net Core SDK for that (e..g `dotnet new console`). To make it easier for you, you can also use [this](https://github.com/hvanbakel/CsprojToVs2017) converter tool

**Comment by edyoung** [comment]
As discussed above, not planning to merge this as-is

**Pull request closed without merging**